pkg/operator/apiserver/controller/workload/workload.go

```diff
@@ -356,6 +424,55 @@ func (c *Controller) updateOperatorStatus(ctx context.Context, previousStatus *o
 	return nil
 }
 
+func (c *Controller) deleteWorkload(ctx context.Context, workloadName string) (err error) {
```

Comment:
Do we need to split it from the `updateOperatorStatus` reconciliation logic? It would be nicer to keep the concern in the same place and consider all 4 conditions.

I do not have a problem with calling a util function in `updateOperatorStatus` if needed though.

Reply:
still applies, if we need to manage the status/conditions during scale down of the workload, it would be better to do it in a single place (updateOperatorStatus)